wolfcrypt/src/port/Renesas/renesas_tsip_aes.c

```diff
@@ -0,0 +1,150 @@
+/* renesas_tsip_aes.c
+ *
+ * Copyright (C) 2006-2019 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <string.h>
+#include <stdio.h>
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+#include <wolfssl/wolfcrypt/settings.h>
+#include <stdio.h>
+
+#ifndef NO_AES
+
+#if defined(WOLFSSL_RENESAS_TSIP_CRYPT) && \
+    !defined(NO_WOLFSSL_RENESAS_TSIP_CRYPT_AES)
+
+#include <wolfssl/wolfcrypt/wc_port.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+
+#include <wolfssl/wolfcrypt/aes.h>
+#include "wolfssl/wolfcrypt/port/Renesas/renesas-tsip-crypt.h"
+
+struct Aes;
+
+int wc_tsip_AesCbcEncrypt(struct Aes* aes, byte* out, const byte* in, word32 sz)
+{
+    tsip_aes_handle_t _handle;
+    word32 ret;
+    word32 blocks = (sz / AES_BLOCK_SIZE);
+    uint32_t dataLength;
+    byte *iv;
+    
+    /* while doing TLS handshake, TSIP driver keeps true-key and iv on the device */
+    /* iv is dummy */
+    iv = (uint8_t*)aes->reg;
```

Comment:
This should go below the NULL checks, in case "aes" is NULL.

Reply:
Thanks!